hw1_classes/code/task1.py

```diff
@@ -0,0 +1,75 @@
+import datetime
+
+
+class Chat:
+    def __init__(self, chat_history=[]):
+        self.chat_history = chat_history
+        self.sent_message_info = None
+
+    def recieve(self):
+        # add datetime of message and put in at start of list chat_history
+        if self.sent_message_info is not None:
+            self.sent_message_info[1] = str(datetime.datetime.now())
+            self.chat_history.insert(0, self.sent_message_info)
+            self.sent_message_info = None
+        else:
+            raise 'The message wasn\'t sent to the chat'
+
+    def show_last_message(self, idx=0):
+        # show last message like - Username (datetime): Text
+        print(f'{self.chat_history[idx][0]} ({self.chat_history[idx][1]}): {self.chat_history[idx][2]}')
```

Comment:
1. Зачем нам индекс, если название метода говорит, что мы показываем **последнее** сообщение?
2. Если бы мы в `chat_history` хранили объекты типа `Message`, то можно было бы просто написать
```suggestion
    def show_last_message(self):
        self.chat_history[0].show()
```